File: twoPointers/containerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxWater, l, r = 0, 0, len(height) - 1
        while l < r:
            maxWater = max(maxWater, (r - l) * min(height[l], height[r]))
            if height[l] > height[r]:
                r -= 1
            else:
                l += 1
        return maxWater

File: twoPointers/containerWithMostWater.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int res = 0;
        int l = 0;
        int r = heights.size() - 1;
        while (l < r){
            int cur_water = min(heights[l], heights[r]) * (r - l);
            res = max(res, cur_water);
            if (heights[l] < heights[r])
                l++;
            else
                r--;
        }
        return res;   
    }
};
